cloudbuild.yaml

```diff
@@ -279,67 +268,77 @@ steps:
   - 'BUILD_ID=$BUILD_ID'
   - 'TRIGGER_NAME=$TRIGGER_NAME'
 
-#
-# Run the e2e tests with FeatureGates inverted compared to Stable
-#
-# Keep in sync with (the inverse of) pkg/util/runtime/features.go:featureDefaults
+# cancel all the orphan e2e test cloud builds, fail to cancel any of the build will fail this whole build
 
-- name: 'e2e-runner'
-  args:
-    - 'CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&SafeToEvict=true&PodHostname=true&SplitControllerAndExtensions=true&Example=true'
-    - 'generic'
-    - 'e2e-test-cluster'
-    - 'us-west1-c'
-    - "${_REGISTRY}"
-  id: e2e-feature-gates
+- name: 'gcr.io/cloud-builders/gcloud'
+  id: cancel-orphan-e2e-tests
   waitFor:
     - e2e-wait-to-become-leader
+  script: |
+    #!/usr/bin/env bash
 
-#
-# Run the e2e tests without FeatureGates
-#
-
-- name: 'e2e-runner'
-  args:
-    - ''
-    - 'generic'
-    - 'e2e-test-cluster'
-    - 'us-west1-c'
-    - "${_REGISTRY}"
-  id: e2e-stable
-  waitFor:
-    - e2e-feature-gates
-
-#
-# Run the e2e tests on GKE Autopilot with FeatureGates inverted compared to Stable
-#
-# Keep in sync with (the inverse of) pkg/util/runtime/features.go:featureDefaults
-
-- name: 'e2e-runner'
-  args:
-    - 'CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&SafeToEvict=true&PodHostname=true&SplitControllerAndExtensions=true&Example=true'
-    - 'gke-autopilot'
-    - 'gke-autopilot-e2e-test-cluster-1-24'
-    - 'us-west1'
-    - "${_REGISTRY}"
-  id: e2e-feature-gates-gke-autopilot
-  waitFor:
-    - e2e-wait-to-become-leader
+    on_going_builds=($(gcloud builds list --ongoing --filter "tags:'e2e-test'" --format="value(id)" | xargs))
+    for on_going_build in ${on_going_builds[@]}
```

Comment:
It looks like `gcloud builds cancel` takes multiple IDs, any reason for the loop?

I think this could all be collapsed into:

```
gcloud builds list --ongoing --filter "tags:'e2e-test'" --format="value(id)" | xargs --no-run-if-empty gcloud builds cancel 
```

Reply:
I was not sure about if there is any limit around the number of IDs `gcloud builds cancel` can take at once.  So just using loop for safety. I will take your suggestion If there is no concern around this, it's much more elegant.

Reply:
The `xargs` default limit is substantial, but more importantly, worst case it just spills to N commands.